src/App.tsx

```diff
@@ -1,61 +1,108 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import React, { useState } from 'react';
+import { Todo } from './types/Todo';
+import { Select } from './components/Select/Select';
+import { Input } from './components/Input/Input';
+import { User } from './types/User';
+
+const getTodoList = todosFromServer.map(todo => ({
+  ...todo,
+  user: usersFromServer.find(u => u.id === todo.userId) || {},
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState<Todo[]>(getTodoList);
+  const [newTodo, setNewTodo] = useState<Todo>({
+    id: 0,
+    title: '',
+    completed: false,
+    userId: 0,
+    user: {},
+  });
+  const [errorTitle, setErrorTitle] = useState('');
+  const [errorUser, setErrorUser] = useState('');
+
+  function addTodo(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+
+    const id = Math.max(...todos.map(u => u.id)) + 1;
```

Comment:
The calculation of `id` for new todos assumes that `todos` is not empty. If `todos` is empty, `Math.max` will return `-Infinity`, resulting in an incorrect ID. Consider handling the case where `todos` is empty by setting `id` to `1` in such cases.